### ISSUE TITLE
legend：how to set dynamic itemWidth？

### ISSUE THREAD

**Issue by wulucxy** [plouc/nivo#253]
![image](https://user-images.githubusercontent.com/5326403/43431196-3b94150c-949f-11e8-99d1-3a54ef8ffa4e.png)

Hi, thanks for this awesome project！

I have some legend labels that with different size，it may looks very ugly when I set the fixed itemWidth，so how can I set dynamic itemWidth?

thanks!

**Comment by borisrorsvort** [comment]
Would it be also possible to make it wrap on multiple lines? 
cf https://github.com/plouc/nivo/issues/274#issuecomment-416024704

**Comment by plouc** [comment]
You cannot as it's not supported by SVG, neither wrap.

**Comment by maximmold** [comment]
Can you use textSpans to break up the text to wrap?

**Comment by plouc** [comment]
Not now, but it could be useful to support some kind of label function which can return any valid SVG.
This is supported for other components/properties ([example](http://nivo.rocks/storybook/?selectedKind=Radar&selectedStory=custom%20label%20component&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs)].

**Comment by cazapatamar** [comment]
@plouc Any update with dynamic itemWidth in Line Chart? Thanks for your answer

**Comment by sattuchin** [comment]
bump

**Comment by tony-scio** [comment]
Anyone have a workaround for this? It's kind of a showstopper when you want a horizontal legend and don't know the width of your labels ahead of time.

**Comment by dkim-thomas** [comment]
@tony-scio 

I went down a small rabbit hole to create dynamic width labels for the Nivo legend. I ultimately decided to just render a legend outside of Nivo as it is way more flexible and less bloated code than my example provided below.  If you want to use the Nivo legend, the code I provided below uses the layer prop to achieve the dynamic widths.

**Code Snippet Example**
https://codesandbox.io/s/zealous-currying-eg5sz?file=/src/App.js
